src/main/java/com/github/xabgesagtx/bots/TelegramBotAutoConfiguration.java

```diff
@@ -31,70 +33,84 @@
 public class TelegramBotAutoConfiguration {
 
 	private List<BotSession> sessions = new ArrayList<>();
-	
+
 	private final List<LongPollingBot> pollingBots;
-	private final List<TelegramWebhookBot> webHookBots;
+	private final List<TelegramWebhookBotService> webHookBots;
 	private final TelegramProperties properties;
+	private SetWebhook.SetWebhookBuilder webhookBuilder;
 
-	static {
-		ApiContextInitializer.init();
-	}
-	
 	@PostConstruct
-	public void start() throws TelegramApiRequestException {
+	public void start() throws TelegramApiException {
 		log.info("Starting auto config for telegram bots");
+		webhookBuilder = SetWebhook.builder();
 		TelegramBotsApi api = telegramBotsApi();
 		pollingBots.forEach(bot -> {
 			try {
 				log.info("Registering polling bot: {}", bot.getBotUsername());
 				sessions.add(api.registerBot(bot));
 			} catch (TelegramApiException e) {
 				log.error("Failed to register bot {} due to error", bot.getBotUsername(), e);
-			}			
+			}
 		});
 		webHookBots.forEach(bot -> {
 			try {
 				log.info("Registering web hook bot: {}", bot.getBotUsername());
-				api.registerBot(bot);
+				SetWebhook webhook = bot.customizeWebHook(webhookBuilder).build();
+				api.registerBot(bot, webhook);
 			} catch (TelegramApiException e) {
 				log.error("Failed to register bot {} due to error", bot.getBotUsername(), e);
-			}			
+			}
 		});
 	}
 
 	/**
 	 * Get API object depending on configured properties.
 	 * If no properties are configured, the API object won't support webhooks.
+	 *
 	 * @return api object
 	 * @throws TelegramApiRequestException in case the creation of the API object fails
 	 */
 	@Bean
 	@ConditionalOnMissingBean
-	public TelegramBotsApi telegramBotsApi() throws TelegramApiRequestException {
+	public TelegramBotsApi telegramBotsApi() throws TelegramApiException {
 		TelegramBotsApi result;
+		DefaultWebhook webhook = getWebhookConfig(properties);
```

Comment:
Minor: I would prefer creating the DefaultWebhook only in case, we create an API with web hook support